src/com/hms/flexyosisoftconnector/DataPoster.java

```diff
@@ -32,134 +32,150 @@ public void run()
+                         case OSIsoftConfig.piwebapi:
+                             OSIsoftServer.addPointToBatch(((Tag) OSIsoftConfig.tags.get(tagIndex)),(DataPoint)((ArrayList)dataPoints.get(tagIndex)).get(dataPointIndex));
+
+                             break;
+                         default:
+                             Logger.LOG_ERR(comsErrMsg);
+                             break;
+                      }
+
+                     pointsAdded++;
+                  }
+                  else Logger.LOG_ERR("Null data point encountered");
+               }
+               switch(communicationType) {
+                  case OSIsoftConfig.omf:
+                      OSIsoftServer.addContainerEndToOMFDataMessage();
+                      break;
+                  case OSIsoftConfig.piwebapi:
+                      OSIsoftServer.startBatch();
+                      break;
+                  default:
+                      Logger.LOG_ERR(comsErrMsg);
+                      break;
                }
-               else Logger.LOG_ERR("Null data point encountered");
-            }
-            switch(communicationType) {
-               case OSIsoftConfig.omf:
-                   OSIsoftServer.addContainerEndToOMFDataMessage();
-                   break;
-               case OSIsoftConfig.piwebapi:
-                   OSIsoftServer.startBatch();
-                   break;
-               default:
-                   Logger.LOG_ERR(comsErrMsg);
-                   break;
             }
-         }
-
-         switch(communicationType) {
-            case OSIsoftConfig.omf:
-                OSIsoftServer.endOMFDataMessage();
-                break;
-            case OSIsoftConfig.piwebapi:
-                OSIsoftServer.endBatch();
-                break;
-            default:
-                Logger.LOG_ERR(comsErrMsg);
-                break;
-         }
-
-
-         if(pointsAdded>0)
-         {
-            Logger.LOG_DEBUG("Sending " + pointsAdded + " points to server");
-            //Post the tags to the server
-            boolean retval = false;
 
             switch(communicationType) {
                case OSIsoftConfig.omf:
-                   retval = OSIsoftServer.postOMFBatch();
+                   OSIsoftServer.endOMFDataMessage();
                    break;
                case OSIsoftConfig.piwebapi:
-                   retval = OSIsoftServer.postBatch();
+                   OSIsoftServer.endBatch();
                    break;
                default:
                    Logger.LOG_ERR(comsErrMsg);
                    break;
             }
 
-            if(retval)
+
+            if(pointsAdded>0)
             {
-               //Remove all the points that were posted
-               for (int tagIndex = 0; tagIndex < OSIsoftConfig.tags.size(); tagIndex++)
+               Logger.LOG_DEBUG("Sending " + pointsAdded + " points to server");
+               //Post the tags to the server
+               boolean retval = false;

```

Comment:
retval doesnt seem like a good name for this variable, since retval isnt a value getting returned.